### PR TITLE
Fix reorg handling

### PR DESCRIPTION
**Pull request by obycode**
This uses @kantai's suggestion to avoid the race condition detected when deciding if a reorg happened.

### CODE REVIEW

File: src/burnchains/mod.rs [hunk 1]
```diff
@@ -476,6 +476,8 @@ pub enum Error {
     TrySyncAgain,
     UnknownBlock(BurnchainHeaderHash),
     CoordinatorClosed,
+    /// SQLite error
+    SQLiteError(rusqlite::Error),
```
Comment (jbencin) [inline src/burnchains/mod.rs:480]: Is this new error type necessary when we already have `DBError`? I'd guess `DBError` has implemented `From<rusqlite::Error>` already, and if not it probably should

File: src/burnchains/mod.rs [hunk 2]
```diff
@@ -495,6 +497,7 @@ impl fmt::Display for Error {
             Error::TrySyncAgain => write!(f, "Try synchronizing again"),
             Error::UnknownBlock(block) => write!(f, "Unknown burnchain block {}", block),
             Error::CoordinatorClosed => write!(f, "ChainsCoordinator channel hung up"),
+            Error::SQLiteError(e) => write!(f, "SQLite error: {}", e),
```
Comment (jbencin) [inline src/burnchains/mod.rs:500]: Minor, but using inline variables looks better:

```suggestion
            Error::SQLiteError(e) => write!(f, "SQLite error: {e}"),
```

**Review by jbencin — APPROVED** [review]
I left a couple minor suggestions but looks good. This should fix the race condition here, but we should think of more general ways to prevent or recover from these situations.

File: src/burnchains/mod.rs (hunk 2 above)
Comment (obycode) [reply src/burnchains/mod.rs:500]: Ah, yeah. I should've learned this by now from your other PRs 😅.

File: src/burnchains/mod.rs (hunk 1 above)
Comment (obycode) [reply src/burnchains/mod.rs:480]: Oh you're right. It does. I'll update to use that.

Comment (obycode) [reply src/burnchains/mod.rs]: Fixed in 8b25efed4e917cbd986b72a258e047696d81f3bd

**Review by kantai — APPROVED** [review]

**Pull request merged**